File: src/main/java/io/github/shirohoo/library/application/persistance/user/UserJpaRepository.java
package io.github.shirohoo.library.application.persistance.user;

import io.github.shirohoo.library.application.domain.user.User;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserJpaRepository extends JpaRepository<User, Long> {
    Optional<User> findByName(String name);
}

File: src/main/kotlin/io/github/shirohoo/library/application/persistence/user/UserJpaRepository.kt
package io.github.shirohoo.library.application.persistence.user

import io.github.shirohoo.library.domain.user.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface UserJpaRepository : JpaRepository<User, Long> {
    fun findByName(name: String): User?

    @Query("select distinct u from User u join fetch u.userLoanHistories")
    fun findAllWithHistories(): List<User>
}